src/credentials/examples/DeviceAttestationVerifierExample.cpp

```diff
@@ -294,6 +328,67 @@ AttestationVerificationResult ExampleDACVerifier::ValidateCertificationDeclarati
     return AttestationVerificationResult::kSuccess;
 }
 
+AttestationVerificationResult
+ExampleDACVerifier::ValidateCertificateDeclarationPayload(const ByteSpan & certDeclBuffer, const ByteSpan & firmwareInfo,
+                                                          uint16_t clusterVendorId, uint16_t clusterProductId, uint16_t dacVendorId,
+                                                          uint16_t dacProductId, uint16_t paiProductId)
+{
+    CertificationElements decodedElements;
+    VerifyOrReturnError(DecodeCertificationElements(certDeclBuffer, decodedElements) == CHIP_NO_ERROR,
+                        AttestationVerificationResult::kCertificationDeclarationInvalidFormat);
+
+    if (!firmwareInfo.empty())
+    {
+        // TODO: check if version_number field in Certification Declaration matches the one in Firmware Information.
+    }
+
+    // The vendor_id field in the Certification Declaration SHALL match the VendorID attribute found in the Basic Information
+    // cluster
+    VerifyOrReturnError(decodedElements.VendorId == clusterVendorId,
+                        AttestationVerificationResult::kCertificationDeclarationInvalidVendorId);
+
+    //  The product_id_array field in the Certification Declaration SHALL contain the value of the ProductID attribute found in the
+    //  Basic Information cluster.
+    VerifyOrReturnError(IsIDContainedInArray(clusterProductId, decodedElements.ProductIds, decodedElements.ProductIdsCount),
+                        AttestationVerificationResult::kCertificationDeclarationInvalidProductId);
+
+    if (decodedElements.DACOriginVIDandPIDPresent)
+    {
+        // The Vendor ID (VID) subject DN in the DAC SHALL match the dac_origin_vendor_id field in the Certification Declaration.
+        VerifyOrReturnError(dacVendorId == decodedElements.DACOriginVendorId,
+                            AttestationVerificationResult::kCertificationDeclarationInvalidVendorId);
+        // The Product ID (PID) subject DN in the DAC SHALL match the dac_origin_product_id field in the Certification Declaration.
+        VerifyOrReturnError(dacProductId == decodedElements.DACOriginProductId,
+                            AttestationVerificationResult::kCertificationDeclarationInvalidProductId);
+        // The Product ID (PID) subject DN in the PAI, if such a Product ID is present, SHALL match the dac_origin_product_id field
+        // in the Certification Declaration.
+        if (paiProductId != 0xFFFFu) // if PAI PID is present
```

Comment:
Product ID 0 is not valid, so absent value is 0 by convention, not FFFF

Reply:
Addressed in d0f7bb1